firebase_admin/_user_mgt.py

```diff
@@ -37,66 +37,79 @@
 ])
 
 
-class _Validator(object):
-    """A collection of data validation utilities.
+class _Unspecified(object):
+    pass
 
-    Methods provided in this class raise ValueErrors if any validations fail. Normal returns
-    signal success.
-    """
+# Use this internally, until sentinels are available in the public API.
+_UNSPECIFIED = _Unspecified()
+
+
+class _Validator(object):
+    """A collection of data validation utilities."""
 
     @classmethod
-    def validate_uid(cls, uid):
+    def validate_uid(cls, uid, required=False):
+        if uid is None and not required:
+            return None
         if not isinstance(uid, six.string_types) or not uid or len(uid) > 128:
             raise ValueError(
                 'Invalid uid: "{0}". The uid must be a non-empty string with no more than 128 '
                 'characters.'.format(uid))
+        return uid
 
     @classmethod
-    def validate_email(cls, email):
-        if not isinstance(email, six.string_types) or not email:
+    def validate_email(cls, email, required=False):
+        if email is None and not required:
+            return None
+        if not isinstance(email, six.string_types):
```

Comment:
Actually an empty string check would be redundant in these cases. For example in case of emails we check:

```
parts = email.split('@')
if len(parts) != 2 or not parts[0] or not parts[1]:
  raise ValueError(...)
```

This fails for empty string. A similar check exists for phone numbers. We also have test coverage for empty email and phone number input.